File: MeuSitePessoal.Domain/Article.cs
namespace MeuSitePessoal.Domain;

public class Article
{
    public Guid Id { get; set; }
    public string Title { get; set; }
    public string Content { get;  set; }
    public string Summary { get; set; }
    public List<string> Tags { get; set; }
    public DateTime CreatedAt { get; set; }

    public Article(string title, string content, string summary, List<string> tags)
    {
        if (string.IsNullOrWhiteSpace(title))
            throw new ArgumentException("The title of the article is required.");
        
        if (string.IsNullOrWhiteSpace(content))
            throw new ArgumentException("The content of the article is required.");

        Id = Guid.NewGuid();
        Title = title;
        Content = content;
        Summary = summary;
        Tags = tags ?? new List<string>();
        CreatedAt = DateTime.UtcNow;
    }
}

File: meupessoal-web/src/models/Article.ts
import { ArticleCategory } from './ArticleCategory';

// Defines the structure of an API article to ensure strong typing in the frontend.
export interface Article {
    id: string;
    title: string;
    titleEn: string;
    titlePt: string;
    content: string;
    contentEn: string;
    contentPt: string;
    summary: string;
    summaryEn: string;
    summaryPt: string;
    tags: string[];
    category: ArticleCategory;
    createdAt: string;
}

// Represents the pagination structure returned by PagedList.cs.
export interface PagedArticles {
    items: Article[];
    currentPage: number;
    totalPages: number;
    pageSize: number;
    totalCount: number;
    hasPrevious: boolean;
    hasNext: boolean;
}